airbyte-server/src/main/java/io/airbyte/server/converters/WorkspaceWebhookConfigsConverter.java

```diff
@@ -10,42 +10,58 @@
+import java.util.ArrayList;
 import java.util.Collections;
+import java.util.Iterator;
 import java.util.List;
 import java.util.UUID;
+import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
+// NOTE: we suppress this warning because PMD thinks it can be a foreach loop in toApiReads but the
+// compiler disagrees.
+@SuppressWarnings("PMD.ForLoopCanBeForeach")
 public class WorkspaceWebhookConfigsConverter {
 
-  public static JsonNode toPersistenceWrite(List<WebhookConfigWrite> apiWebhookConfigs) {
+  public static JsonNode toPersistenceWrite(List<WebhookConfigWrite> apiWebhookConfigs, Supplier<UUID> uuidSupplier) {
     if (apiWebhookConfigs == null) {
       return Jsons.emptyObject();
     }
 
     final WebhookOperationConfigs configs = new WebhookOperationConfigs()
-        .withWebhookConfigs(apiWebhookConfigs.stream().map(WorkspaceWebhookConfigsConverter::toPersistenceConfig).collect(Collectors.toList()));
+        .withWebhookConfigs(apiWebhookConfigs.stream().map((item) -> toPersistenceConfig(uuidSupplier, item)).collect(Collectors.toList()));
 
     return Jsons.jsonNode(configs);
   }
 
-  public static List<WebhookConfigRead> toApiReads(List<WebhookConfig> persistenceConfig) {
-    if (persistenceConfig.isEmpty()) {
+  public static List<WebhookConfigRead> toApiReads(final JsonNode persistenceConfig) {
+    if (persistenceConfig == null) {
       return Collections.emptyList();
     }
-    return persistenceConfig.stream().map(WorkspaceWebhookConfigsConverter::toApiRead).collect(Collectors.toList());
+
+    // NOTE: we deserialize it "by hand" because the secrets aren't hydrated, so we can't deserialize it
+    // into the usual shape.
+    // TODO(mfsiega-airbyte): find a cleaner way to handle this situation.
+    List<WebhookConfigRead> configReads = new ArrayList<>();
+    final JsonNode configArray = persistenceConfig.findPath("webhookConfigs");
+    for (Iterator<JsonNode> it = configArray.elements(); it.hasNext();) {
+      JsonNode webhookConfig = it.next();
+      configReads.add(toApiRead(webhookConfig));
+    }
+    return configReads;
   }
 
-  private static WebhookConfig toPersistenceConfig(final WebhookConfigWrite input) {
+  private static WebhookConfig toPersistenceConfig(final Supplier<UUID> uuidSupplier, final WebhookConfigWrite input) {
     return new WebhookConfig()
-        .withId(UUID.randomUUID())
+        .withId(uuidSupplier.get())
         .withName(input.getName())
         .withAuthToken(input.getAuthToken());
   }
 
-  private static WebhookConfigRead toApiRead(final WebhookConfig persistenceConfig) {
+  private static WebhookConfigRead toApiRead(final JsonNode configJson) {
     final var read = new WebhookConfigRead();
-    read.setId(persistenceConfig.getId());
-    read.setName(persistenceConfig.getName());
+    read.setId(UUID.fromString(configJson.findValue("id").asText()));
+    read.setName(configJson.findValue("name").asText());
```

Comment:
one thought: I'm curious if it's possible to reference this to the annotation name so it doesn't out of sync. I can't think of an easy way to do so so maybe not worth doing. Just throwing it out there.

Reply:
I did think about trying to inspect the schema, but couldn't really figure out anything that felt valuable. The one positive here is that if the schema changes in a breaking way and we forget to update this, the workspace handler unit tests will fail (I double checked the test coverage report, we are indeed hitting these lines) - so I'm okay with this for now.

(In the future, I do think this deserves revisiting.)

Reply:
Nice!